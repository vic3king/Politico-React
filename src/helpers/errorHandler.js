const errorHandler = obj => {
  let error;

  if (typeof obj === 'string') {
    error = obj;
  }

  if (typeof obj === 'object') {
    Object.values(obj).forEach(err => {
      error = err;
    });
  }

  return error;
};

export default errorHandler;
