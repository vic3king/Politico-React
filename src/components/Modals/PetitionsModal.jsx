import React, { Component } from 'react';
import { notify } from 'react-notify-toast';
import Loader from '../shared/Loader/Loader';
import Petitions from '../../services/petitions';
import errorHandler from '../../helpers/errorHandler';
import '../../style/admin.scss';

class PetitionsModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      formDetails: {},
      loading: false,
    };
  }

  hide = e => {
    e.preventDefault();
    const { hide } = this.props;
    hide();
  };

  onInputChange = event => {
    const { formDetails } = this.state;
    formDetails[event.target.id] = event.target.value;
    this.setState({ formDetails });
  };

  onButtonSubmit = async event => {
    event.preventDefault();
    this.setState({ loading: true });
    const { hide } = this.props;
    const { formDetails } = this.state;
    formDetails.office = Number(formDetails.office);

    const petition = await Petitions.postPetition(formDetails);

    if (petition.status >= 400) {
      this.setState({ loading: false });
      notify.show(errorHandler(petition.error.message[0].error), 'error');
    }

    if (petition.status === 201) {
      this.setState({ loading: false });
      hide();
      notify.show('success');
    }
  };

  render() {
    const { formDetails, loading } = this.state;
    const { offices } = this.props;

    return (
      <React.Fragment>
        {loading && <Loader />}
        <div>
          <section className="modal-main">
            <form className="modal-form">
              <h3 className="page-modal-title">Petition</h3>
              <div className="fix">
                <span>Select an office to petition</span>
                <select id="office" onChange={this.onInputChange}>
                  <option disabled selected default>
                    -- Select Office --
                  </option>
                  {offices.map(obj => (
                    <option key={obj.id} name={obj.name} value={obj.id}>
                      {obj.name}
                    </option>
                  ))}
                </select>
              </div>
              <div className="fix">
                <span>Submit any piece of evidence</span>
                <input
                  type="text"
                  id="evidence"
                  placeholder="Enter link here"
                  value={formDetails.name}
                  onChange={this.onInputChange}
                  required
                />
              </div>
              <div className="fix">
                <span>Comments</span>
                <textarea
                  rows="4"
                  cols="50"
                  placeholder="Leave your comments here"
                  id="comment"
                  onChange={this.onInputChange}
                />
              </div>
              <button
                id="office-btn"
                type="submit"
                className="cte-off"
                onClick={this.onButtonSubmit}
              >
                Submit
              </button>
              <button type="button" id="btn" onClick={this.hide}>
                X
              </button>
            </form>
          </section>
        </div>
      </React.Fragment>
    );
  }
}

export default PetitionsModal;
