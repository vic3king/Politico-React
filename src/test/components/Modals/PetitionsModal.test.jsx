import React from 'react';
import { shallow } from 'enzyme';
import { notify } from 'react-notify-toast';
import PetitionsModal from '../../../components/Modals/PetitionsModal';
import Petitions from '../../../services/petitions';
import errorHandler from '../../../helpers/errorHandler';

jest.mock('../../../services/petitions');
jest.mock('react-notify-toast');

const hide = jest.fn();
const offices = [
  {
    name: '',
  },
  {
    name: '',
  },
];

let wrapper;
describe('PetitionsModal component', () => {
  beforeEach(() => {
    wrapper = shallow(<PetitionsModal hide={hide} offices={offices} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('hide method', () => {
    wrapper = shallow(<PetitionsModal hide={hide} offices={offices} />);
    const instance = wrapper.instance();
    let event;
    it('should call preventDefault on event', async () => {
      event = {
        preventDefault: jest.fn(),
      };

      await instance.hide(event);
      expect(event.preventDefault).toBeCalled();
    });
  });

  describe('onInputChange method', () => {
    it('should update formData state ', () => {
      const instance = wrapper.instance();
      const event = {
        preventDefault: jest.fn(),
        target: { id: 'office' },
      };
      const formDetails = {
        name: 'test',
      };
      instance.setState({
        formDetails,
      });

      instance.onInputChange(event);

      expect(instance.state.formDetails).toEqual({
        name: 'test',
      });
    });
  });

  describe('onButtonSubmit method', () => {
    let instance;
    let event;

    describe('Api call success', () => {
      beforeAll(() => {
        Petitions.postPetition = jest.fn().mockImplementation(() =>
          Promise.resolve({
            status: 201,
          })
        );
      });
      beforeEach(() => {
        instance = wrapper.instance();
        event = {
          preventDefault: jest.fn(),
        };
      });

      it('should call preventDefault on event', async () => {
        await instance.onButtonSubmit(event);

        expect(event.preventDefault).toBeCalled();
      });
      it('should call toast the right message on api call success', async () => {
        await instance.onButtonSubmit(event);
        expect(notify.show).toBeCalledWith('success');
      });
      it('should set loading state to false after successfull api call is made', async () => {
        await instance.onButtonSubmit(event);
        expect(instance.state.loading).toBeFalsy();
      });
      it('should call the offices method with id from api call', async () => {
        await instance.onButtonSubmit(event);
        expect(hide).toBeCalled();
      });
    });

    describe('Api call 400 errors', () => {
      beforeAll(() => {
        Petitions.postPetition = jest.fn().mockImplementation(() =>
          Promise.resolve({
            status: 400,
            error: {
              message: [{ error: 'validation error' }],
            },
          })
        );
      });
      beforeEach(() => {
        instance = wrapper.instance();
        event = {
          preventDefault: jest.fn(),
        };
      });

      it('should set loading state to false on validation error', async () => {
        await instance.onButtonSubmit(event);
        expect(instance.state.loading).toBeFalsy();
      });

      it('should toast the error message', async () => {
        await instance.onButtonSubmit(event);
        expect(notify.show).toBeCalledWith(
          errorHandler('validation error'),
          'error'
        );
      });
    });
  });
});
