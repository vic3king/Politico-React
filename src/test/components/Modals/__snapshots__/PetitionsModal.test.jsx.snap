// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionsModal component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PetitionsModal
    hide={[MockFunction]}
    offices={
      Array [
        Object {
          "name": "",
        },
        Object {
          "name": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <$class
          options={Object {}}
        />,
        false,
        <div>
          <section
            className="modal-main"
          >
            <form
              className="modal-form"
            >
              <h3
                className="page-modal-title"
              >
                Petition
              </h3>
              <div
                className="fix"
              >
                <span>
                  Select an office to petition
                </span>
                <select
                  id="office"
                  onChange={[Function]}
                >
                  <option
                    name=""
                  >
                    
                  </option>
                  <option
                    name=""
                  >
                    
                  </option>
                </select>
              </div>
              <div
                className="fix"
              >
                <span>
                  Submit any piece of evidence
                </span>
                <input
                  id="evidence"
                  onChange={[Function]}
                  placeholder="Enter link here"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="fix"
              >
                <span>
                  Comments
                </span>
                <textarea
                  cols="50"
                  id="comment"
                  onChange={[Function]}
                  placeholder="Leave your comments here"
                  rows="4"
                />
              </div>
              <button
                className="cte-off"
                id="office-btn"
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>
              <button
                id="btn"
                onClick={[Function]}
                type="button"
              >
                X
              </button>
            </form>
          </section>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "options": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="modal-main"
          >
            <form
              className="modal-form"
            >
              <h3
                className="page-modal-title"
              >
                Petition
              </h3>
              <div
                className="fix"
              >
                <span>
                  Select an office to petition
                </span>
                <select
                  id="office"
                  onChange={[Function]}
                >
                  <option
                    name=""
                  >
                    
                  </option>
                  <option
                    name=""
                  >
                    
                  </option>
                </select>
              </div>
              <div
                className="fix"
              >
                <span>
                  Submit any piece of evidence
                </span>
                <input
                  id="evidence"
                  onChange={[Function]}
                  placeholder="Enter link here"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="fix"
              >
                <span>
                  Comments
                </span>
                <textarea
                  cols="50"
                  id="comment"
                  onChange={[Function]}
                  placeholder="Leave your comments here"
                  rows="4"
                />
              </div>
              <button
                className="cte-off"
                id="office-btn"
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>
              <button
                id="btn"
                onClick={[Function]}
                type="button"
              >
                X
              </button>
            </form>
          </section>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="modal-form"
            >
              <h3
                className="page-modal-title"
              >
                Petition
              </h3>
              <div
                className="fix"
              >
                <span>
                  Select an office to petition
                </span>
                <select
                  id="office"
                  onChange={[Function]}
                >
                  <option
                    name=""
                  >
                    
                  </option>
                  <option
                    name=""
                  >
                    
                  </option>
                </select>
              </div>
              <div
                className="fix"
              >
                <span>
                  Submit any piece of evidence
                </span>
                <input
                  id="evidence"
                  onChange={[Function]}
                  placeholder="Enter link here"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="fix"
              >
                <span>
                  Comments
                </span>
                <textarea
                  cols="50"
                  id="comment"
                  onChange={[Function]}
                  placeholder="Leave your comments here"
                  rows="4"
                />
              </div>
              <button
                className="cte-off"
                id="office-btn"
                onClick={[Function]}
                type="submit"
              >
                Submit
              </button>
              <button
                id="btn"
                onClick={[Function]}
                type="button"
              >
                X
              </button>
            </form>,
            "className": "modal-main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="page-modal-title"
                >
                  Petition
                </h3>,
                <div
                  className="fix"
                >
                  <span>
                    Select an office to petition
                  </span>
                  <select
                    id="office"
                    onChange={[Function]}
                  >
                    <option
                      name=""
                    >
                      
                    </option>
                    <option
                      name=""
                    >
                      
                    </option>
                  </select>
                </div>,
                <div
                  className="fix"
                >
                  <span>
                    Submit any piece of evidence
                  </span>
                  <input
                    id="evidence"
                    onChange={[Function]}
                    placeholder="Enter link here"
                    required={true}
                    type="text"
                  />
                </div>,
                <div
                  className="fix"
                >
                  <span>
                    Comments
                  </span>
                  <textarea
                    cols="50"
                    id="comment"
                    onChange={[Function]}
                    placeholder="Leave your comments here"
                    rows="4"
                  />
                </div>,
                <button
                  className="cte-off"
                  id="office-btn"
                  onClick={[Function]}
                  type="submit"
                >
                  Submit
                </button>,
                <button
                  id="btn"
                  onClick={[Function]}
                  type="button"
                >
                  X
                </button>,
              ],
              "className": "modal-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Petition",
                  "className": "page-modal-title",
                },
                "ref": null,
                "rendered": "Petition",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Select an office to petition
                    </span>,
                    <select
                      id="office"
                      onChange={[Function]}
                    >
                      <option
                        name=""
                      >
                        
                      </option>
                      <option
                        name=""
                      >
                        
                      </option>
                    </select>,
                  ],
                  "className": "fix",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Select an office to petition",
                    },
                    "ref": null,
                    "rendered": "Select an office to petition",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <option
                          name=""
                        >
                          
                        </option>,
                        <option
                          name=""
                        >
                          
                        </option>,
                      ],
                      "id": "office",
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "name": "",
                          "value": undefined,
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "name": "",
                          "value": undefined,
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "option",
                      },
                    ],
                    "type": "select",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Submit any piece of evidence
                    </span>,
                    <input
                      id="evidence"
                      onChange={[Function]}
                      placeholder="Enter link here"
                      required={true}
                      type="text"
                    />,
                  ],
                  "className": "fix",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Submit any piece of evidence",
                    },
                    "ref": null,
                    "rendered": "Submit any piece of evidence",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "evidence",
                      "onChange": [Function],
                      "placeholder": "Enter link here",
                      "required": true,
                      "type": "text",
                      "value": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Comments
                    </span>,
                    <textarea
                      cols="50"
                      id="comment"
                      onChange={[Function]}
                      placeholder="Leave your comments here"
                      rows="4"
                    />,
                  ],
                  "className": "fix",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Comments",
                    },
                    "ref": null,
                    "rendered": "Comments",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "cols": "50",
                      "id": "comment",
                      "onChange": [Function],
                      "placeholder": "Leave your comments here",
                      "rows": "4",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "cte-off",
                  "id": "office-btn",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "id": "btn",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
            ],
            "type": "form",
          },
          "type": "section",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <$class
            options={Object {}}
          />,
          false,
          <div>
            <section
              className="modal-main"
            >
              <form
                className="modal-form"
              >
                <h3
                  className="page-modal-title"
                >
                  Petition
                </h3>
                <div
                  className="fix"
                >
                  <span>
                    Select an office to petition
                  </span>
                  <select
                    id="office"
                    onChange={[Function]}
                  >
                    <option
                      name=""
                    >
                      
                    </option>
                    <option
                      name=""
                    >
                      
                    </option>
                  </select>
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Submit any piece of evidence
                  </span>
                  <input
                    id="evidence"
                    onChange={[Function]}
                    placeholder="Enter link here"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Comments
                  </span>
                  <textarea
                    cols="50"
                    id="comment"
                    onChange={[Function]}
                    placeholder="Leave your comments here"
                    rows="4"
                  />
                </div>
                <button
                  className="cte-off"
                  id="office-btn"
                  onClick={[Function]}
                  type="submit"
                >
                  Submit
                </button>
                <button
                  id="btn"
                  onClick={[Function]}
                  type="button"
                >
                  X
                </button>
              </form>
            </section>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "options": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [MockFunction],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="modal-main"
            >
              <form
                className="modal-form"
              >
                <h3
                  className="page-modal-title"
                >
                  Petition
                </h3>
                <div
                  className="fix"
                >
                  <span>
                    Select an office to petition
                  </span>
                  <select
                    id="office"
                    onChange={[Function]}
                  >
                    <option
                      name=""
                    >
                      
                    </option>
                    <option
                      name=""
                    >
                      
                    </option>
                  </select>
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Submit any piece of evidence
                  </span>
                  <input
                    id="evidence"
                    onChange={[Function]}
                    placeholder="Enter link here"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Comments
                  </span>
                  <textarea
                    cols="50"
                    id="comment"
                    onChange={[Function]}
                    placeholder="Leave your comments here"
                    rows="4"
                  />
                </div>
                <button
                  className="cte-off"
                  id="office-btn"
                  onClick={[Function]}
                  type="submit"
                >
                  Submit
                </button>
                <button
                  id="btn"
                  onClick={[Function]}
                  type="button"
                >
                  X
                </button>
              </form>
            </section>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="modal-form"
              >
                <h3
                  className="page-modal-title"
                >
                  Petition
                </h3>
                <div
                  className="fix"
                >
                  <span>
                    Select an office to petition
                  </span>
                  <select
                    id="office"
                    onChange={[Function]}
                  >
                    <option
                      name=""
                    >
                      
                    </option>
                    <option
                      name=""
                    >
                      
                    </option>
                  </select>
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Submit any piece of evidence
                  </span>
                  <input
                    id="evidence"
                    onChange={[Function]}
                    placeholder="Enter link here"
                    required={true}
                    type="text"
                  />
                </div>
                <div
                  className="fix"
                >
                  <span>
                    Comments
                  </span>
                  <textarea
                    cols="50"
                    id="comment"
                    onChange={[Function]}
                    placeholder="Leave your comments here"
                    rows="4"
                  />
                </div>
                <button
                  className="cte-off"
                  id="office-btn"
                  onClick={[Function]}
                  type="submit"
                >
                  Submit
                </button>
                <button
                  id="btn"
                  onClick={[Function]}
                  type="button"
                >
                  X
                </button>
              </form>,
              "className": "modal-main",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="page-modal-title"
                  >
                    Petition
                  </h3>,
                  <div
                    className="fix"
                  >
                    <span>
                      Select an office to petition
                    </span>
                    <select
                      id="office"
                      onChange={[Function]}
                    >
                      <option
                        name=""
                      >
                        
                      </option>
                      <option
                        name=""
                      >
                        
                      </option>
                    </select>
                  </div>,
                  <div
                    className="fix"
                  >
                    <span>
                      Submit any piece of evidence
                    </span>
                    <input
                      id="evidence"
                      onChange={[Function]}
                      placeholder="Enter link here"
                      required={true}
                      type="text"
                    />
                  </div>,
                  <div
                    className="fix"
                  >
                    <span>
                      Comments
                    </span>
                    <textarea
                      cols="50"
                      id="comment"
                      onChange={[Function]}
                      placeholder="Leave your comments here"
                      rows="4"
                    />
                  </div>,
                  <button
                    className="cte-off"
                    id="office-btn"
                    onClick={[Function]}
                    type="submit"
                  >
                    Submit
                  </button>,
                  <button
                    id="btn"
                    onClick={[Function]}
                    type="button"
                  >
                    X
                  </button>,
                ],
                "className": "modal-form",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Petition",
                    "className": "page-modal-title",
                  },
                  "ref": null,
                  "rendered": "Petition",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Select an office to petition
                      </span>,
                      <select
                        id="office"
                        onChange={[Function]}
                      >
                        <option
                          name=""
                        >
                          
                        </option>
                        <option
                          name=""
                        >
                          
                        </option>
                      </select>,
                    ],
                    "className": "fix",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Select an office to petition",
                      },
                      "ref": null,
                      "rendered": "Select an office to petition",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <option
                            name=""
                          >
                            
                          </option>,
                          <option
                            name=""
                          >
                            
                          </option>,
                        ],
                        "id": "office",
                        "onChange": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "",
                            "name": "",
                            "value": undefined,
                          },
                          "ref": null,
                          "rendered": "",
                          "type": "option",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "",
                            "name": "",
                            "value": undefined,
                          },
                          "ref": null,
                          "rendered": "",
                          "type": "option",
                        },
                      ],
                      "type": "select",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Submit any piece of evidence
                      </span>,
                      <input
                        id="evidence"
                        onChange={[Function]}
                        placeholder="Enter link here"
                        required={true}
                        type="text"
                      />,
                    ],
                    "className": "fix",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Submit any piece of evidence",
                      },
                      "ref": null,
                      "rendered": "Submit any piece of evidence",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "evidence",
                        "onChange": [Function],
                        "placeholder": "Enter link here",
                        "required": true,
                        "type": "text",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Comments
                      </span>,
                      <textarea
                        cols="50"
                        id="comment"
                        onChange={[Function]}
                        placeholder="Leave your comments here"
                        rows="4"
                      />,
                    ],
                    "className": "fix",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Comments",
                      },
                      "ref": null,
                      "rendered": "Comments",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "cols": "50",
                        "id": "comment",
                        "onChange": [Function],
                        "placeholder": "Leave your comments here",
                        "rows": "4",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "textarea",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "cte-off",
                    "id": "office-btn",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "id": "btn",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            "type": "section",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
